infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -223,6 +228,15 @@
                         .setNameFormat(serverContext.getThreadPrefix() + "LocalMetricsPolling")
                         .build());
 
+        // Creating a cached thread pool to spawn a task to bootstrap the sequencer. The number of
+        // tasks submitted to this thread pool is limited to one. We check if we have an ongoing
+        // task before we launch a new one.
+        this.sequencerBootstrapWorker = Executors.newCachedThreadPool(
```

Comment:
This class has too many thread pools, why is this needed?  Why not reuse the detectionTaskWorkers pool? Also, why is this a cched thread pool, if all you want to do is cancel a task, you can use a single threaded executor.

Reply:
I can reuse the detectionWorkers pool by supplying it an extra thread.
I used the cached thread pool to optimize for thread resources and return the thread once the task is completed.
In a good scenario where we have minimal epoch changes, this thread will be wasted.

Reply:
Optimize as needed. 

Are you saying that newCachedThreadPool will destroy the thread if its not being used?

Reply:
I see. I don't think this optimization is necessary.

Reply:
Ok, I'll add a thread to detectionWorkersPool then.

Reply:
@Maithem If we are going to increase the detection thread pool size why not just using SingleThreadExecutor? @zalokhan Although cached thread pool shed their non-core threads, the Executor's default has unlimited max. to take the same benefit maybe consider:
`this.sequencerBootstrapWorker = new ThreadPoolExecutor(0, 1,  
                                                      60, TimeUnit.SECONDS, 
                                                       aQueue, aThreadFactory); `

It has an upper limit of 1 which is semantically inline with logic of the bootstrapping (no more than 1 bootstrapping task) and it has 0 core thread.